src/module.c

```diff
@@ -606,6 +615,17 @@ client *moduleAllocTempClient(user *user) {
     return c;
 }
 
+static void freeRedisModuleAsyncRMCallPromise(RedisModuleAsyncRMCallPromise *promise) {
+    if (--promise->ref_count > 0) {
+        return;
+    }
+    /* When the promise is finally freed it can not have a client attached to it.
+     * Either releasing the client would have remove it.
+     * Or RM_CallReplyPromiseAbort would have remove it. */
```

Comment:
i think either-or should be one sentence.
```suggestion
     * Either releasing the client or RM_CallReplyPromiseAbort would have removed it. */
```